Bottom: b2f530b8a202a5c8793debe55accaab26daf9829
Top:    c968af53ccaf837d2b2b7634561f8f05ececd4a4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 14:07:26 +0100

Revert "hid/gtk: Add hooks to tell the render when we start/stop drawing"

This reverts commit 332cb8ddc1cc032ac29c7d2b9a0051ccc228b6da.


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 3d90c27..90cbdc0 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1035,17 +1035,6 @@ ghid_drawing_area_configure_hook (GHidPort *port)
     }
 }
 
-gboolean
-ghid_start_drawing (GHidPort *port)
-{
-  return TRUE;
-}
-
-void
-ghid_end_drawing (GHidPort *port)
-{
-}
-
 void
 ghid_screen_update (void)
 {
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 70342d4..d3d5c16 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -655,13 +655,8 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
 
-
   gdk_event_request_motions (ev);
 
-  if (!ghid_start_drawing (out))
-    return FALSE;
-
-
   if (out->panning)
     {
       dx = gport->zoom * (x_prev - ev->x);
@@ -683,8 +678,6 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   ghid_show_crosshair (TRUE);
   if (moved && have_crosshair_attachments ())
     ghid_draw_area_update (gport, NULL);
-
-  ghid_end_drawing (out);
   return FALSE;
 }
 
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 5a51358..d4fc8fa 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -500,8 +500,6 @@ void ghid_show_crosshair (gboolean show);
 void ghid_init_renderer (int *, char ***, GHidPort *);
 void ghid_init_drawing_widget (GtkWidget *widget, GHidPort *);
 void ghid_drawing_area_configure_hook (GHidPort *port);
-gboolean ghid_start_drawing (GHidPort *port);
-void ghid_end_drawing (GHidPort *port);
 void ghid_screen_update (void);
 gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
                                       GHidPort *);
